gcs/constraints/smart_table.cc

```diff
@@ -50,32 +54,20 @@ namespace
     using TreeEdges = vector<vector<SmartEntry>>;
     using Forest = vector<TreeEdges>;
 
-    // --- remove eventually -- DEBUGGING ONLY
-    //    auto index_of(const IntegerVariableID & val, const vector<IntegerVariableID> & vec) -> int
-    //    {
-    //        ptrdiff_t pos = distance(vec.begin(), find(vec.begin(), vec.end(), val));
-    //        return (int)pos;
-    //    }
-
-    // // -- remove eventually -- DEBUGGING ONLY
-    //    auto print_edge(const SmartEntry & edge, const vector<IntegerVariableID> vars) -> void
-    //    {
-    //        const string stringtypes[] = {"LESS_THAN", "LESS_THAN_EQUAL", "EQUAL", "NOT_EQUAL", "GREATER_THAN", "GREATER_THAN_EQUAL", "IN", "NOT_IN"};
-    //        overloaded{
-    //            [&](BinaryEntry b) {
-    //                cout << index_of(b.var_1, vars) << ", " << stringtypes[b.constraint_type] << ", " << index_of(b.var_2, vars);
-    //            },
-    //            [&](UnarySetEntry us) {
-    //                cout << &us.var;
-    //            },
-    //            [&](UnaryValueEntry us) {
-    //                cout << &us.var;
-    //            }}
-    //            .visit(edge);
-    //    }
-    // // ---
-
-    auto filter_edge(const SmartEntry & edge, VariableDomainMap & supported_by_tree) -> void
+    auto log_filtering_inference(const ProofFlag & tuple_selector, const Literal & lit, State & state)
+    {
+        state.infer(TrueLiteral{}, JustifyExplicitly{[&](Proof & proof, vector<ProofLine> &) -> void {
```

Comment:
Compiler warning, should check state.infer's return value and throw an exception if it's not Unchanged.